src/util/GeoSparqlHelpers.cpp

```diff
@@ -13,64 +15,85 @@ namespace ad_utility {
 
 namespace detail {
 
-// Helper function that checks whether a string looks like a WKT pointC (just
-// checks the beginning and that it has the minimal length).
-bool isWktPoint(const std::string& p) {
-  return p.size() >= 12 && p[0] == '"' && (p[1] == 'P' || p[1] == 'p') &&
-         (p[2] == 'O' || p[2] == 'o') && (p[3] == 'I' || p[3] == 'i') &&
-         (p[4] == 'N' || p[4] == 'n') && (p[5] == 'T' || p[5] == 't') &&
-         p[6] == '(';
+// Helper function that checks whether the given null-terminated string is a WKT
+// point according to the official grammar. If yes, return pointer to first
+// character after opening bracket, otherwise nullptr. Anything after the final
+// quote is ignored for now.
+const char* checkWktPoint(const char* s) {
+  bool isPrefixOk =
+      s[0] == '"' && (s[1] == 'P' || s[1] == 'p') &&
+      (s[2] == 'O' || s[2] == 'o') && (s[3] == 'I' || s[3] == 'i') &&
+      (s[4] == 'N' || s[4] == 'n') && (s[5] == 'T' || s[5] == 't');
+  s += 6;
+  if (!isPrefixOk) return nullptr;
+  while (*s == ' ') ++s;
+  if (*s++ != '(') return nullptr;
+  while (*s == ' ') ++s;
+  const char* beg = s;
+  for (int i = 1; i <= 2; ++i) {
+    while (*s == ' ') ++s;
+    if (*s == '-') ++s;
+    if (!(isdigit(*s++))) return nullptr;
+    while (isdigit(*s)) ++s;
+    if (*s == '.') {
+      ++s;
+      if (!(isdigit(*s++))) return nullptr;
+      while (isdigit(*s)) ++s;
+    }
+  }
+  while (*s == ' ') ++s;
+  if (*s++ != ')') return nullptr;
+  if (*s++ != '"') return nullptr;
```

Comment:
Ok, since you are doing the parsing very explicitly now, isn't this basically a regex?

Reply:
See here for an example (missing: the second coordinate and the skipping of whitespace).

Reply:
[Here is the link I have forgotten](https://godbolt.org/z/W3xe4eTz5)

Reply:
Thanks for the hint, Johannes! The code is so much better now. What was I even thinking?